### ISSUE TITLE
[BUG] Running example throws "RuntimeError: grad can be implicitly created only for scalar outputs"

### ISSUE THREAD

**Issue by LMKight** [NVIDIA-Merlin/Transformers4Rec#592]
### Bug description
Trying to run the example throws the following exception when training is started 
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File <timed exec>:18

File /usr/local/lib/python3.8/dist-packages/transformers/trainer.py:1316, in Trainer.train(self, resume_from_checkpoint, trial, ignore_keys_for_eval, **kwargs)
   1314         tr_loss_step = self.training_step(model, inputs)
   1315 else:
-> 1316     tr_loss_step = self.training_step(model, inputs)
   1318 if (
   1319     args.logging_nan_inf_filter
   1320     and not is_torch_tpu_available()
   1321     and (torch.isnan(tr_loss_step) or torch.isinf(tr_loss_step))
   1322 ):
   1323     # if loss is nan or inf simply add the average of previous logged losses
   1324     tr_loss += tr_loss / (1 + self.state.global_step - self._globalstep_last_logged)

File /usr/local/lib/python3.8/dist-packages/transformers/trainer.py:1867, in Trainer.training_step(self, model, inputs)
   1865     loss = self.deepspeed.backward(loss)
   1866 else:
-> 1867     loss.backward()
   1869 return loss.detach()

File /usr/local/lib/python3.8/dist-packages/torch/_tensor.py:402, in Tensor.backward(self, gradient, retain_graph, create_graph, inputs)
    393 if has_torch_function_unary(self):
    394     return handle_torch_function(
    395         Tensor.backward,
    396         (self,),
   (...)
    400         create_graph=create_graph,
    401         inputs=inputs)
--> 402 torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)

File /usr/local/lib/python3.8/dist-packages/torch/autograd/__init__.py:184, in backward(tensors, grad_tensors, retain_graph, create_graph, grad_variables, inputs)
    180 inputs = (inputs,) if isinstance(inputs, torch.Tensor) else \
    181     tuple(inputs) if inputs is not None else tuple()
    183 grad_tensors_ = _tensor_or_tensors_to_tuple(grad_tensors, len(tensors))
--> 184 grad_tensors_ = _make_grads(tensors, grad_tensors_, is_grads_batched=False)
    185 if retain_graph is None:
    186     retain_graph = create_graph

File /usr/local/lib/python3.8/dist-packages/torch/autograd/__init__.py:85, in _make_grads(outputs, grads, is_grads_batched)
     83 if out.requires_grad:
     84     if out.numel() != 1:
---> 85         raise RuntimeError("grad can be implicitly created only for scalar outputs")
     86     new_grads.append(torch.ones_like(out, memory_format=torch.preserve_format))
     87 else:

RuntimeError: grad can be implicitly created only for scalar outputs
```

### Steps/Code to reproduce bug
1. docker pull nvcr.io/nvidia/merlin/merlin-pytorch:22.11
2. Start docker container as described here: https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/main/examples
3. Clone the repo into the docker container
4. in the 2nd notebook comment out the data_loader_engine='merlin' because this doesn't work
5. execute the first 2 notebooks in examples/getting-started-session-based

### Expected behavior
The example runs through without crashing

### Environment details
- Transformers4Rec version: 0.1.15
- Platform: docker / ubuntu
- Python version: 3.8.10
- Huggingface Transformers version: 4.12.0
- PyTorch version: 1.12.1 with GPU
- Tensorflow version (GPU?): -
- CUDA version: 11.7

### Additional context
I commented out the data_loader_engine='merlin' param in the config because this throws an error that merlin was never registered with pytorch

**Comment by LMKight** [comment]
I tried it with an older version of the example now and that worked - it works with commit 6582b65a (v0.1.15)

Usually i assume that the main branch is on the same version as the latest release, maybe you can consider using a dev branch for unreleased changes.

**Comment by rnyak** [comment]
@LMKight same issue was reported here https://github.com/NVIDIA-Merlin/Transformers4Rec/issues/583 and looks like they solved it with  python 3.7.12.

**Comment by rnyak** [comment]
closing since there is no further activitiy.
